ui/components/component-library/button-icon/button-icon.types.ts

```diff
@@ -0,0 +1,49 @@
+import type { BoxProps } from '../../ui/box/box.d';
+import { IconName, IconSize } from '../icon';
+import { Color, IconColor } from '../../../helpers/constants/design-system';
+
+export enum ButtonIconSize {
+  Sm = IconSize.Sm,
+  Lg = IconSize.Lg,
+}
+
+export interface ButtonIconProps extends BoxProps {
+  /**
+   * String that adds an accessible name for ButtonIcon
+   */
+  ariaLabel: string;
+  /**
+   * The polymorphic `as` prop allows you to change the root HTML element of the Button component between `button` and `a` tag
+   */
+  as?: keyof JSX.IntrinsicElements;
+  /**
+   * An additional className to apply to the ButtonIcon.
+   */
+  className?: string;
+  /**
+   * The color of the ButtonIcon component should use the IconColor object from
+   * ./ui/helpers/constants/design-system.js
+   */
+  color?: IconColor | Color;
```

Comment:
Let's only allow `IconColor` we don't allow any custom icons or to allow for the icon to be a letter right?
```suggestion
  color?: IconColor;
```